opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -203,6 +203,10 @@ type TlsConfigTransport struct {
 	NodesDn []string `json:"nodesDn,omitempty"`
 	// DNs of certificates that should have admin access, mainly used for securityconfig updates via securityadmin.sh, only used when existing certificates are provided
 	AdminDn []string `json:"adminDn,omitempty"`
+	// When re-generating certificate: deadline before regenerating, in days. Empty/zero means 30 days
+	RegenDeadlineDays int `json:"regenDeadlineDays,omitempty"`
+	// When generating certificate: certificate validity in days. Empty/zero means 1 year
+	ValidityDays int `json:"validityDays,omitempty"`
```

Comment:
I suggest to not introduce these changes in criteo forks, we are fine with hardcoded value for now and it will reduce risk of conflicts with upstream. You will propose them when preparing upstream contribution after having taken maintainer advise.